### ISSUE TITLE
The `removeUnusedKeys` option re-orders keys in translation file

### ISSUE THREAD

**Issue by bbthorntz** [i18next/i18next-scanner#64]
Using the `removeUnusedKeys` is re-ordering many items in my translation file. This makes it very difficult to diff which translations were added/removed. Disabling this option works as expected.

#### Version
- i18next: <i>10.2.1</i>
- i18next-scanner: <i>2.2.1</i>

#### Configuration
```js
{
    options: {
        debug: true,
        removeUnusedKeys: true,
        sort: false,
        attr: false,
        func: {
            list: ['__'],
            extensions: ['.js', '.jsx']
        },
        ns: ['common'],
        defaultNs: 'common',
        nsSeparator: '::',
        keySeparator: false,
        contextFallback: false,
        plural: false,
        pluralFallback: false,
        interpolation: {
            prefix: '{{',
            suffix: '}}',
        },
        lngs: ['ja'],
        defaultValue: null,
        resource: {
            loadPath: 'i18n/{{lng}}/{{ns}}.json',
            savePath: 'i18n/{{lng}}/{{ns}}.json',
            jsonIndent: 4,
        },
    }
};
```

**Comment by cheton** [comment]
Hi @bbthorntz 

I may not have time to change the code recently, would you mind to create a PR to fix this issue?

The code change should be around the following lines:

https://github.com/i18next/i18next-scanner/blob/master/src/parser.js#L485-L492
```js
if (opts.sort) {
    // Sort object keys in ascending order
    //    :   :   :
} else if (resStore !== this.resStore) {
    // Add your code here
    // Sort "resStore" according to the object key order in "this.resStore"
}
```

**Comment by bbthorntz** [comment]
Thanks @cheton. Sorry for the slow reply. Will try make some time for this this week.

**Comment by cheton** [comment]
For a short term solution, you can turn on the "sort" option to deeply sort the resStore object, it's the most easy way to have ordered keys.

**Comment by cheton** [comment]
Fixed in [v2.4.2](https://github.com/i18next/i18next-scanner/releases/tag/v2.4.2) with commit a6fa06c9acd84198fb8f8c19bc0ba177c390f4c4
```js
if (this.options.removeUnusedKeys) {
    // Merge two objects `resStore` and `resScan` deeply, returning a new merged object with the elements from both `resStore` and `resScan`.
    const resMerged = deepMerge(this.resStore, this.resScan);
    Object.keys(this.resStore).forEach((lng) => {
        Object.keys(this.resStore[lng]).forEach((ns) => {
            const resStoreKeys = flattenObjectKeys(_.get(this.resStore, [lng, ns], {}));
            const resScanKeys = flattenObjectKeys(_.get(this.resScan, [lng, ns], {}));
            const unusedKeys = _.differenceWith(resStoreKeys, resScanKeys, _.isEqual);
            for (let i = 0; i < unusedKeys.length; ++i) {
                _.unset(resMerged[lng][ns], unusedKeys[i]);
            }
            // Omit empty object
            resMerged[lng][ns] = omitEmptyObject(resMerged[lng][ns]);
        });
    });
    resStore = resMerged;
}
```
